internal/controller/resourcesetinputprovider_controller.go

```diff
@@ -219,28 +226,42 @@ func (r *ResourceSetInputProviderReconciler) reconcile(ctx context.Context,
 		providerCtx, cancel := context.WithTimeout(ctx, obj.GetTimeout())
 		defer cancel()
 
-		// Create the provider based on the object type.
-		provider, err := r.newGitProvider(providerCtx, obj, certPool, authData)
-		if err != nil {
-			msg := fmt.Sprintf("failed to create provider %s", err.Error())
-			conditions.MarkFalse(obj,
-				meta.ReadyCondition,
-				meta.ReconciliationFailedReason,
-				"%s", msg)
-			r.notify(ctx, obj, corev1.EventTypeWarning, meta.ReconciliationFailedReason, msg)
-			return ctrl.Result{}, err
-		}
+		switch {
+		case strings.HasPrefix(obj.Spec.Type, "Git"):
```

Comment:
This will break #302 

We now have 2 level deep switches, let's create a function `reconcileExternalProvider` that returns the inputs.

Reply:
Thanks! It made more sense to call it `callExternalProvider` instead and return just the exported inputs instead of a reconcile result. We were handling all the errors in this part the same way, so now we just return the error in `callExternalProvider` and then update the conditions and call `r.notify()` in the caller.